### DISCUSSION TITLE
Enabling 'Push to Talk' or 'Detect Voice' mode from a non-local connection? (Multiuser mode)

### DISCUSSION THREAD

**Discussion by cgessai** [LostRuins/koboldcpp#914] [General]
I love being able to speak to my LLM model via the WhisperCPP support.  I want to have that same functionality when my Android phone connects to my computer running KoboldCPP in Multiuser Mode on my local network.  However, when I try it a dialog window pops up on my phone saying 'Cannot initialize microphone.  If you're using a non-localhost URL, it needs to be served over HTTPS!'  

What's the minimum series of steps/fewest settings to enable another device on my local network to access either the 'Push to Talk' or 'Detect Voice' functionality?   Any help is appreciated!  I assume it's possible?

**Comment by LostRuins** [comment]
If you want to use microphone on a different host, it must be served over HTTPS. Assuming you don't want to use the `--remotetunnel` option, then you need to setup a SSL cert and load that from koboldcpp. The instructions for that are on the koboldcpp wiki
